src/System.Net.Security/src/System/Net/SslStreamPal.Unix.cs

```diff
@@ -131,13 +131,17 @@ private static SecurityStatusPal EncryptDecryptHelper(SafeDeleteContext security
             try
             {
                 Interop.Ssl.SslErrorCode errorCode = Interop.Ssl.SslErrorCode.SSL_ERROR_NONE;
-
-
                 SafeSslHandle scHandle = securityContext.SslContext;
 
-                resultSize = encrypt ?
-                    Interop.OpenSsl.Encrypt(scHandle, buffer, offset, size, out errorCode) :
-                    Interop.OpenSsl.Decrypt(scHandle, buffer, size, out errorCode);
+                if (encrypt)
+                {
+                    resultSize = Interop.OpenSsl.Encrypt(scHandle, buffer, offset, size, out errorCode);
+                }
+                else
+                {
+                    Debug.Assert(offset == 0, "Expected offset 0 when decrypting");
```

Comment:
The two eventual call sites always explicitly pass 0.  The call paths are a bit windy; it'd be nice to clean it up at some point to avoid offset even getting passed in.